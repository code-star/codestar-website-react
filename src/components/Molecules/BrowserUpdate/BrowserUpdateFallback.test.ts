import addBrowserUpdate from './BrowserUpdateFallback';

const globalAny: any = global;

jest.useFakeTimers();

describe('addBrowserUpdate()', () => {
  beforeEach(() => {
    document.body.innerHTML = '<div id="dummy-content"></div>';
  });

  describe('Snaphot', () => {
    it('shows message if Map.prototype.keys not defined', () => {
      const originalMapKeys = globalAny.Map.prototype.keys;
      delete Map.prototype.keys;
      expect(addBrowserUpdate('dummy-content')).toMatchSnapshot();
      jest.advanceTimersByTime(1000);
      expect(document.body).toMatchSnapshot();
      globalAny.Map.prototype.keys = originalMapKeys;
    });

    it('does not show message if Map.prototype.keys is defined', () => {
      expect(addBrowserUpdate('dummy-content')).toMatchSnapshot();
      expect(document.body).toMatchSnapshot();
    });
  });
});
