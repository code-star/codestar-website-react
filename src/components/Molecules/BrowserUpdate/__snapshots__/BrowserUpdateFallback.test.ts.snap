// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addBrowserUpdate() Snaphot does not show message if Map.prototype.keys is defined 1`] = `undefined`;

exports[`addBrowserUpdate() Snaphot does not show message if Map.prototype.keys is defined 2`] = `
<body>
  <div
    id="dummy-content"
  />
</body>
`;

exports[`addBrowserUpdate() Snaphot shows message if Map.prototype.keys not defined 1`] = `
<div
  style="
    background-color: white;
    border: 0.5em solid red;
    font-size: 200%;
    font-weight: bold;
    min-height: 6em;
    margin-top: 4rem;
    padding: 1em;
    text-align: center;
  "
>
  <p>
    Please upgrade your browser.
  </p>
  <p>
    <a
      href="https://browser-update.org/update-browser.html"
    >
      More information
    </a>
  </p>
</div>
`;

exports[`addBrowserUpdate() Snaphot shows message if Map.prototype.keys not defined 2`] = `
<body>
  <div
    style="
    background-color: white;
    border: 0.5em solid red;
    font-size: 200%;
    font-weight: bold;
    min-height: 6em;
    margin-top: 4rem;
    padding: 1em;
    text-align: center;
  "
  >
    <p>
      Please upgrade your browser.
    </p>
    <p>
      <a
        href="https://browser-update.org/update-browser.html"
      >
        More information
      </a>
    </p>
  </div>
  <div
    id="dummy-content"
  />
</body>
`;
